junit-platform-commons-java-9/src/main/java/org/junit/platform/commons/util/JigsawUtils.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.io.IOException;
+import java.lang.module.ModuleFinder;
+import java.lang.module.ModuleReader;
+import java.lang.module.ModuleReference;
+import java.lang.module.ResolvedModule;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.JUnitException;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+
+/**
+ * Collection of utilities for working with {@code java.lang.Module}
+ * and friends.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public class JigsawUtils {
```

Comment:
This class is the only mention of "Jigsaw" in the codebase. I'm wondering whether `ModuleUtils` would be a better name... Thoughts? 🤔

Reply:
I intend to refactor the `eval()/invokeGetters()` logic used in `TestEngine` to a call to `JigsawUtils`. Something like `JigsawUtils.getModuleName(Class<?> type)` and `JigsawUtils.getModuleVersion(Class<?> type)` returning `Optional<String>` of course.

Reply:
Does that speak against `ModuleUtils`?

Reply:
Looking at the proposed names ... no. Not yet. :)